app/searchForm/index.js

```diff
@@ -0,0 +1,34 @@
+class searchFormController {
+    /*@ngInject*/
+    constructor(gitSearchService) {
+        this.query = '';
+        this.gitSearchService = gitSearchService;
+    }
+
+    getRequestParams() {
+        let parsedParams = {};
+        let params = this.query.split('+');
+        params.forEach((parameter) => {
+            parameter = parameter.trim();
+            let subParams = parameter.split(':');
+            if (subParams.length > 1) {
+                let trimmedParams = subParams.map(Function.prototype.call, String.prototype.trim);
```

Comment:
How it works:
It calls Function.prototype.call with trim as *this* and following arguments passed by map to callback (value, index, array):
Function.prototype.call.call(String.prototype.trim, val, ind, arr)